be/src/storage/lake/tablet_manager.cpp

```diff
@@ -492,6 +503,46 @@ StatusOr<TxnLogIter> TabletManager::list_txn_log(int64_t tablet_id, bool filter_
 }
 
 StatusOr<TabletSchemaPtr> TabletManager::get_tablet_schema(int64_t tablet_id, int64_t* version_hint) {
+// TODO: Eliminate the explicit dependency on staros worker
+#ifdef USE_STAROS
+    auto shard_info_or = g_worker->get_shard_info(tablet_id);
+    if (!shard_info_or.ok()) {
+#ifndef BE_TEST
+        return to_status(shard_info_or.status());
```

Comment:
need fallback to as if the indexId can't be found from shardInfo?

Reply:
Yes, the processing logic is in the else code block below.

Reply:
I mean if the status is not ok(), may be no need to return here. Can still try to go with the cache/listing behavior.

Reply:
updated